common/src/x509.rs

```diff
@@ -0,0 +1,166 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    x509.rs
+
+Abstract:
+
+    File contains X509 Certificate & CSR related utility functions
+
+--*/
+use crate::crypto::{self, PubKey};
+use caliptra_drivers::*;
+use core::mem::size_of;
+use core::usize;
+use zerocopy::AsBytes;
+
+/// X509 API
+pub enum X509 {}
+
+impl X509 {
```

Comment:
Since these are already in a separate `mod`, it feels redundant to have an empty `enum` to contain. We could just have these be functions in this file without the `enum` and `impl`.

That way later we don't have to have the triply redundant `caliptra_x509::x509::X509::method_name()`.

Reply:
Updated.